### PR TITLE
Search: cancel operation if the query is empty

### PR DESCRIPTION
**Pull request by RDIL**
Signed-Off-By: Reece Dunham <me@rdil.rocks>

Don't continue the search if there is no query (just for performance reasons so this logic doesn't actually need to be run)

### CODE REVIEW

File: packages/file-search/src/node/file-search-service-impl.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export class FileSearchServiceImpl implements FileSearchService {
         @inject(RawProcessFactory) protected readonly rawProcessFactory: RawProcessFactory) { }
 
     async find(searchPattern: string, options: FileSearchService.Options, clientToken?: CancellationToken): Promise<string[]> {
+        if (searchPattern === "") {
```
Comment (vince-fugnitto) [inline packages/file-search/src/node/file-search-service-impl.ts]: @RDIL please use singlequotes for strings (reason for the build failure).

https://github.com/eclipse-theia/theia/blob/86704a9ca6106ca5c9c34f7b42c6066b82cb1d82/configs/errors.eslintrc.json#L7-L13

Comment (RDIL) [reply packages/file-search/src/node/file-search-service-impl.ts]: Ah got it

Comment (RDIL) [reply packages/file-search/src/node/file-search-service-impl.ts]: Fixed.

**Comment by vince-fugnitto** [comment]
@RDIL please take a look at the failing tests.
You can execute the tests like so (from the command line):

```
npx run test @theia/file-search
```

**Comment by westbury** [comment]
@RIDL Can you detail how you are doing the search?  If I use Ctrl-P and type stuff, it does not send through anything to the modified code until the first character is typed.  Also I don't see any performance issue if I clear out the search.

**Comment by RDIL** [comment]
Oh I'm an idiot, this is the wrong search module.  Sorry

**Pull request closed without merging**